x-pack/plugins/cases/server/services/user_actions/index.test.ts

```diff
@@ -1578,6 +1587,383 @@ describe('CaseUserActionService', () => {
           ]
         `);
       });
+
+      describe('Decode', () => {
```

Comment:
In my opinion this seems like a lot of work to test the transform and decode logic. If you'd like to keep them that's fine. I feel like it'd be more straightforward and readable if we test the functionality directly rather than going through these top level functions 🤷‍♂️. I can see the benefit of exercising the whole flow but it seems like we have to mock a whole lot just to get to that point.